api/rql/internal/predicate/time.go

```diff
@@ -0,0 +1,121 @@
+package predicate
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/puppetlabs/wash/api/rql"
+	"github.com/puppetlabs/wash/api/rql/internal/errz"
+	"github.com/puppetlabs/wash/api/rql/internal/matcher"
+	"github.com/puppetlabs/wash/munge"
+)
+
+func Time(op ComparisonOp, t time.Time) rql.TimePredicate {
+	return &tm{
+		op: op,
+		t:  t,
+	}
+}
+
+// tm => time. Have to name it as such to avoid conflicting with
+// the 'time' package
+type tm struct {
+	op ComparisonOp
+	t  time.Time
```

Comment:
So this goes with absolute time rather than `time.Duration` from a reference point?

Reply:
Yes. I feel like that's more idiomatic with other filtering languages (e.g. I found "return log files that were modified after time X" to be more common).

`wash find` can still implement is past/future time semantics. We'd just create a wrapper to a TimePredicate that would add `[AND, [<=, <ref_time>], <time_predicate>]` to whatever predicate was parsed. Negation can also do a similar thing.

Reply:
If past/future time semantics become common in other front-end interfaces, RQL could probably be updated to support that so people don't have to keep writing wrappers around it.

Reply:
I think that makes sense.